common/djangoapps/student/management/commands/change_enrollment.py

```diff
@@ -59,36 +76,70 @@ class Command(BaseCommand):
         )
-        if options['user']:
-            user_str = options['user']
-            for username in user_str.split(","):
-                if '@' in username:
-                    user = User.objects.get(email=username)
-                else:
-                    user = User.objects.get(username=username)
-                filter_args['user'] = user
-                enrollments = CourseEnrollment.objects.filter(**filter_args)
-                if options['noop']:
-                    print "Would have changed {num_enrollments} students from {from_mode} to {to_mode}".format(
-                        num_enrollments=enrollments.count(),
-                        from_mode=options['from_mode'],
-                        to_mode=options['to_mode']
-                    )
-                else:
+
+        if options['username']:
+            self.update_enrollments('username', enrollment_args, options, error_users, success_users)
+
+        if options['email']:
+            self.update_enrollments('email', enrollment_args, options, error_users, success_users)
+
+        self.report(error_users, success_users)
+
+    def update_enrollments(self, identifier, enrollment_args, options, error_users, success_users):
+        """ Update enrollments for a specific user identifier (email or username). """
+        users = options[identifier].split(",")
+        for identified_user in users:
+            logger.info(identified_user)
+            try:
+                user_args = {
+                    identifier: identified_user
+                }
+
+                enrollment_args['user'] = User.objects.get(**user_args)
+                enrollments = CourseEnrollment.objects.filter(**enrollment_args)
+
+                with transaction.atomic():
                     for enrollment in enrollments:
                         enrollment.update_enrollment(mode=options['to_mode'])
                         enrollment.save()
+
+                    if options['noop']:
+                        raise RollbackException('Forced rollback.')
+
+            except RollbackException:
+                success_users.append(identified_user)
+                continue
+            except Exception as exception:
+                error_users.append((identified_user, exception))
+                continue
+
+            success_users.append(identified_user)
+            logger.info('Updated user [%s] to mode [%s]', identified_user, options['to_mode'])
+
+    def report(self, error_users, success_users):
+        """ Log and overview of the results of the command. """
+        total_users = len(success_users) + len(error_users)
+        logger.info('Successfully updated %i out %i users', len(success_users), total_users)
+        if len(error_users) > 0:
+            logger.info('The following %i user(s) not saved:', len(error_users))
+            for user, error in error_users:
+                logger.info('user: [%s] reason: [%s] %s', user, type(error).__name__, error.message)
```

Comment:
Similar question about logging here -- these messages are just for the person running the command, right?